dev-cpp/folly/folly-2023.01.16.00-r1.ebuild

```diff
@@ -85,6 +85,8 @@ src_configure() {
 		-DBUILD_TESTS=$(usex test)
 	)
 
+	use ppc64 && mycmakeargs+=( -DCMAKE_LIBRARY_ARCHITECTURE=ppc64 )
```

Comment:
Can you please:
1. Make this generic by mapping Gentoo arches to CMake arches? (I'm sure others ebuilds do this)
2. Send a patch upstream to Folly. CMake should be doing the detection for this. It works fine in other software.

Reply:
Regarding 1. this is exactly what I'm asking in my comment. I grepped `DCMAKE_LIBRARY_ARCHITECTURE` and no other ebuild makes use of it. I'm not sure of cmake's naming convention for archs nor how to reliably convert Gentoo's arches into cmake archs because I couldn't find an example in another ebuild either.

About 2. I will eventually try to submit a patch upstream.

Reply:
Upstream is aware of the issue and I've already asked them if they adhere to any specific naming convention: https://github.com/facebook/folly/issues/1851
Even if I manage to get a fix upstream that won't work for already released versions like 2023.01.16.00.

Reply:
I can't know about links and bugs unless you show me them. In future, please include all possible references up front.

I'm not asking about the convention - I'm saying something is going wrong with CMake not automatically detecting.

Obviously a proper fix could be backported to existing versions fine, but they also release weekly, so...

Reply:
@thesamesam I couldn't find a way to get CMake populate `CMAKE_LIBRARY_ARCHITECTURE` and sadly many other projects went the route of hardcoding it to x86. I've looked at their `CMakeLists.txt` and they really only seems to care for x86_64 anyway, so we're basically free to set `CMAKE_LIBRARY_ARCHITECTURE` to whatever we want for any other platform and we would be fine. So I've hardcoded it to `x86_64` for `amd64` and to `$ARCH` for everything else. What do you think about this? I fear I won't be able to get any better solution to this.

Reply:
No problem, I understand. That sounds good enough for now, let's try it.